### PR TITLE
[SLICING] PBI - 22 : Create Lesson Material

### LINKED ISSUE #63: PBI-22: Teacher can create lesson material in class
**Story Point: 8** 

- [x] List of topic (dropdown)
- [x] Title form (Max 255) **(Required)**
- [x] Description form (Maxx 65535)
- [x] File content (Max 1 file)
- [x] Link content (Max 1 link)
- [x] Create Button
- [x] Cancel Button
- [x] Only teacher can create lesson material

### PR DESCRIPTION
**Pull request by fxdros**

### CODE REVIEW

File: src/resources/app/components/Layout/CreationLayout.vue [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<template>
+  <v-app>
+    <v-container>
+      <div class="mt-16">
+        <v-btn
+          id="back"
+          color="white"
+          elevation="0"
+          style="padding-left: 0; padding-right: 8px"
+          @click="redirectBackToClasswork()">
```
Comment (AldiNFitrah) [inline src/resources/app/components/Layout/CreationLayout.vue]:
```suggestion
          @click="redirectBackToClasswork()"
        >
```
To make the hierarchy clearer

File: src/resources/app/components/Layout/CreationLayout.vue [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          class="text-primary text-lg font-medium mb-4 mt-1"
+        >
+          {{pageTitle}}
+        </div>
+        <v-divider id="divider" color="#06BBD3" class="mb-6"></v-divider>
+        <slot/>
+      </div>
+    </v-container>
+  </v-app>
+
```
Comment (AldiNFitrah) [inline src/resources/app/components/Layout/CreationLayout.vue]: no need extra line

**Review by AldiNFitrah — APPROVED** [review]

File: src/resources/app/pages/course/CreateCourseMaterial.vue [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import AppLayout from "../../components/Layout/AppLayout";
+import CreationLayout from "../../components/Layout/CreationLayout";
+
+export default {
+  components: {CreationLayout},
+  layout: [AppLayout],
+  name: "CreateCourseMaterial",
+  data() {
+    return {
+      items: [
```
Comment (Zsaschz) [inline src/resources/app/pages/course/CreateCourseMaterial.vue]: Why its still using static data? I think there is already an api for getting list of topics

File: src/resources/app/pages/course/CreateCourseMaterial.vue [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    </div>
+  </CreationLayout>
+</template>
+
+<script>
+import AppLayout from "../../components/Layout/AppLayout";
+import CreationLayout from "../../components/Layout/CreationLayout";
+
+export default {
+  components: {CreationLayout},
```
Comment (Zsaschz) [inline src/resources/app/pages/course/CreateCourseMaterial.vue]: Add space or make the curly braces on a separate line

**Review by Zsaschz — COMMENTED** [review]
I think for create material page you need to add course id props. This id then can be used to fetch all of the topics for assigning the material

**Review by kanziebub — APPROVED** [review]
what Fathan said, other than that, LGTM!

File: src/resources/app/pages/course/CreateCourseMaterial.vue [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    return {
+      items: []
+    }
+  },
+  methods: {
+    async getTopic() {
+      try {
+        const response = await getTopicAPI(this.$props.id);
+        this.items = response.data;
+      } catch (error) {
```
Comment (Zsaschz) [inline src/resources/app/pages/course/CreateCourseMaterial.vue:86]: please use `this.$store.dispatch("OPEN_SNACKBAR", error message)`

**Review by Zsaschz — APPROVED** [review]
you can use toast for error, other than that it's ok. good job mate!

**Pull request merged**